nntrainer/layers/lstm.cpp

```diff
@@ -218,11 +223,118 @@ void LSTMLayer::copy(std::shared_ptr<Layer> l) {
 }
 
 void LSTMLayer::calcDerivative() {
-  // NYI
+  Tensor &derivative_ = fgio->getGradientRef();
```

Comment:
can you add a check here that this calculation was done in calcGradient()?
If an LSTM layer is set to be non-trainable, then this will break.

Reply:
I checked this is done in `calcGradient()` I think if there is a dependency between those two, everything should be inside `calcDerivative`?

Reply:
Please add a TODO here to add a check later to support  calcDerivative when this layer is non-trainable,

Reply:
Added as a new issue #1226 as discussed offline.